proposals/shared-everything-threads/Overview.md

```diff
@@ -589,6 +589,26 @@ entire thread or agent.
 [jspi]: https://github.com/WebAssembly/js-promise-integration
 [realm]: https://tc39.es/ecma262/#realm
 
+#### Thread-Bound Data
+
+Just as WebAssembly produces expect to be able to call JS functions, they also expect to be able to
+hold handles to JS objects. For example, a language compiled to shared WasmGC may need to hold
+references to DOM nodes in some of its shared objects. Normally this would be disallowed because DOM
+nodes are not shareable and shared-to-unshared references are disallowed. To bridge this gap, we
+need a utility that allows shared-to-unshared objects as an exception to the general rule, assuming
+that Web GCs can support such an exception (see the discussion in the FAQ below). To avoid
+complicating the core language and forcing every non-Web engine to support shared-to-unshared edges,
+this utility is a new JS API.
+
+`ThreadBoundData` is a shared JS object that wraps any other arbitrary JS object. `ThreadBoundData`
+provides access to the wrapped object via a `get` method that throws an exception when called on any
+thread besides the thread that owns the wrapped object, ensuring that the wrapped object can be
```

Comment:
When we say "thread" here, do we really just mean web workers? It might be worthwhile clarifying that...

Reply:
+1 we should be more precise. In specese that would be "agent".

Reply:
Is a function called _on_ an agent or called _by_ an agent?